src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -485,93 +578,255 @@ CHIP_ERROR SilabsTracer::TraceBufferFlushByOperation(size_t aOperation)
+    {
+        return GetTraceByOperation(to_underlying(operationKey), buffer);
+    }
+
+    // Check if it is a named trace
+    const char * base = aOperation.data();
+    const void * pos  = memchr(base, ':', aOperation.size());
+
+    if (pos == nullptr)
+    {
+        ChipLogError(DeviceLayer, "Invalid Trace Operation format");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    const char * colon = static_cast<const char *>(pos);
+
+    size_t groupLen   = static_cast<size_t>(colon - base);
+    size_t labelStart = groupLen + 1;
+
+    if (labelStart > aOperation.size())
+    {
+        ChipLogError(DeviceLayer, "Invalid Trace Operation format");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    size_t labelLen = aOperation.size() - labelStart;
+
+    groupLen = std::min(groupLen, static_cast<size_t>(NamedTrace::kMaxGroupLength > 0 ? NamedTrace::kMaxGroupLength - 1 : 0));
+    labelLen = std::min(labelLen, static_cast<size_t>(NamedTrace::kMaxLabelLength > 0 ? NamedTrace::kMaxLabelLength - 1 : 0));
+
+    CharSpan groupSpan = aOperation.SubSpan(0, groupLen);
+    CharSpan labelSpan = aOperation.SubSpan(labelStart, labelLen);
+
+    size_t mIndex = 0;
+    ReturnErrorOnFailure(FindExistingTrace(labelSpan, groupSpan, mIndex));
+
+    return GetTraceByOperation(mIndex + kNumTraces, buffer);
 }
 
-CHIP_ERROR SilabsTracer::RegisterAppTimeTraceOperation(CharSpan & appOperationKey)
+CHIP_ERROR SilabsTracer::FindOrCreateTrace(const CharSpan label, const CharSpan group, size_t & outIdx)
 {
-    VerifyOrReturnError(appOperationKey.size() <= kMaxAppOperationKeyLength, CHIP_ERROR_BUFFER_TOO_SMALL);
-    VerifyOrReturnError(mAppOperationKeyCount < kMaxAppOperationKeys, CHIP_ERROR_NO_MEMORY);
-    VerifyOrReturnError(appOperationKey.data() != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    // Verify the key is not already in the array
-    size_t index = 0;
-    VerifyOrReturnError(CHIP_ERROR_NOT_FOUND == FindAppOperationIndex(appOperationKey, index), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(FindExistingTrace(label, group, outIdx) != CHIP_NO_ERROR, CHIP_NO_ERROR);
 
-    MutableCharSpan newAppKey(mAppOperationKeys[mAppOperationKeyCount]);
-    CopyCharSpanToMutableCharSpan(appOperationKey, newAppKey);
-    mAppOperationKeyCount++;
-    return CHIP_NO_ERROR;
+    // Find first empty slot
+    for (size_t i = 0; i < kMaxNamedTraces; i++)
+    {
+        if (mNamedTraces[i].labelLen == 0)
+        {
+            auto & trace    = mNamedTraces[i];
+            size_t labelLen = label.size();
+            size_t groupLen = group.size();
+
+            // Truncate label and group if too long
+            if (labelLen >= NamedTrace::kMaxLabelLength)
+                labelLen = NamedTrace::kMaxLabelLength - 1;
+
+            if (groupLen >= NamedTrace::kMaxGroupLength)
+                groupLen = NamedTrace::kMaxGroupLength - 1;
+
+            memcpy(trace.label, label.data(), labelLen);
+            trace.label[labelLen] = '\0';
```

Comment:
I might have missed it, but do we still used std string calls that relly on null-termination? Since Charspan carry their size, they do not need it, but if we still have a need for those null terminators we should document it.